src/ape/utils.py

```diff
@@ -444,6 +445,21 @@ def get_all_files_in_directory(path: Path) -> List[Path]:
     return [path]
 
 
+class AbstractBaseModel(ABC, BaseModel):
+    class Config:
+        keep_untouched = (cached_property,)
+        arbitrary_types_allowed = True
+
+    # NOTE: Due to https://github.com/samuelcolvin/pydantic/issues/1241
+    #       we have to add this cached property workaround in order to avoid this error:
+    #
+    #           TypeError: cannot pickle '_thread.RLock' object
+
+    def __dir__(self) -> List[str]:
```

Comment:
Add note about how this integrates with IPython: https://ipython.readthedocs.io/en/stable/config/integrating.html